### ISSUE TITLE
Autorest (modeler four?) is incorrectly assuming that omitting "type" implies "type": "object"

### ISSUE THREAD

**Issue by johanste** [Azure/autorest#3817]
When you don't provide a "type" attribute for a model, modeler four is unhappy and tries to "fix" it by assuming that it is an "object". However, this is (to the best of my understanding) incorrect. Not including a "type" means that there is no validation/assertion as to what the type is.  

```bash
$> autorest --input-file=any.json --python --verbose

WARNING (PreCheck/SchemaMissingType): The schema 'Any' with an undefined type and additionalProperties is a bit ambigious. This has been auto-corrected to 'type:object'
```
See below for example swagger (yes, I know the same file exposes more than one issue :))

```json
{
  "swagger": "2.0",
  "info": {
    "title": "Recursive",
    "version": "1.0"
  },
  "produces": [ "application/json" ],
  "consumes": [ "application/json" ],
  "paths": {
    "/resources/{resourceId}": {
      "parameters": [
        {
          "name": "resourceId",
          "in": "path",
          "type": "string",
          "required": true
        }
      ],
      "put": {
        "operationId": "createResource",
        "parameters": [
          {
            "name": "body",
            "in": "body",
            "schema": {
              "$ref": "#/definitions/Resource"
            }
          }
          ],
        "responses": {
          "200": {
            "description": "Success, ja!",
            "schema": {
              "$ref": "#/definitions/Resource"
            }
          }
        }
      }
    }
  },
  "definitions": {
    "Any": {
      "additionalProperties": {
        "$ref": "#/definitions/Any"
      }
    },
    "Resource": {
      "type": "object",
      "properties": {
        "name": {
          "type": "string"
        },
        "randomProperties": {
          "$ref": "#/definitions/Any"
        }
      }
    }
  }
}
```

**Comment by timotheeguerin** [comment]
Yeah looking at it, not including the type would mean that this is `any` type. I am guessing this was done for a few reason:
- Helping teams migrate with missing types.
- Not sure we support having a `any` type. It's something we'd have to discuss

**Comment by timotheeguerin** [comment]
However because this is used with `additionalProperties` which means that it is a `Dictionnary` as far as I understand and should have` type:"object"`https://swagger.io/docs/specification/data-models/dictionaries/?sbsearch=additionalProperties

**Comment by johanste** [comment]
The example above is, in fact, trying to explore what the correct way (in swagger 2.0) is to add an actual `any` type. `additionalProperties` only applies if the type is an object, but it shouldn't imply that the type can *only* be an object. As with other validators, they only apply if the type matches what the validator applies to. 

Now, swagger 2.0 has some consistency issues (from what I can tell) in how it deals with what the default value for `additionalProperties` is (it is, per my understanding, effectively `"additionalProperties": false` - but that breaks a lot of stuff in actuality. For example, `allOf` becomes almost useless with that interpretation). OpenApi 3 should have fixed that by more closely conform to json schema (default value is `"additionalProperties": true).

**Comment by timotheeguerin** [comment]
Actually we do support any, there is an `AnySchema` in m4, it can just be done without providing anything on it.

```json
{
  "swagger": "2.0",
  "info": {
    "title": "Recursive",
    "version": "1.0"
  },
  "produces": ["application/json"],
  "consumes": ["application/json"],
  "paths": {},
  "definitions": {
    "Any": {
      
    }
  }
}

```

It is however still logging a warning 

WARNING (Modeler/MissingType): The schema 'Any' has no type or format information whatsoever. Location:
   file:///C:/dev/azsdk/playground/swagger.json#/components/schemas/Any

**Comment by johanste** [comment]
Yeah, that should work - at least in openapi 3. It swagger 2.0, it was much more ambiguous as to what this means...
